app/formations/[id]/edit/page.tsx

```diff
@@ -0,0 +1,46 @@
+"use server";
+
+import Builder from "@/components/Builder";
+import { getCmsData } from "@/lib/server/cms-data";
+import { getFormation } from "@/lib/server/formations";
+import { auth } from "@clerk/nextjs/server";
+
+type Props = {
+  params: { id: string };
+};
+export default async function FormationEditPage({ params }: Props) {
+  const { id } = params;
+  const formation = await getFormation(id);
+  if (!formation) {
+    return (
+      <div className="container md:w-[40vw] text-center">
+        <h1 className="text-2xl font-bold">Formation not found</h1>
+        <p>Check the URL</p>
+      </div>
+    );
+  }
+
+  const { userId } = auth();
+
+  if (!userId) {
+    auth().redirectToSignIn();
+    return;
+  }
+
+  if (formation.user_id !== userId) {
```

Comment:
Do you need this? Shouldn't we put this stuff in middleware to catch for specific pages if they aren't logged in? The only case here would be if the `userId` that you're logged in as isn't the owner of the formation, then we should redirect from the component?

Reply:
i couldn't find good documentation on the clerk side about how to catch dynamic route segments in their middleware.

this particular check is for when someone doesn't own the component.
we can redirect away from the page for sure.

Reply:
I guess my thought is if they hit the `*/edit` route and aren't logged in, they shouldn't even get this far to the component. The middleware would stop this.

Reply:
[clerk docs](https://clerk.com/docs/references/nextjs/clerk-middleware#protect-routes)